engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/generic/ReadArrayElementNode.java

```diff
@@ -13,32 +13,34 @@
 import org.enso.interpreter.runtime.error.PanicException;
 
 @BuiltinMethod(
-        type = "Polyglot",
-        name = "read_array_element",
-        description = "Returns the size of a polyglot array.")
+    type = "Polyglot",
+    name = "read_array_element",
+    description = "Returns the size of a polyglot array.")
 public class ReadArrayElementNode extends Node {
-    private @Child
-    InteropLibrary library =
-            InteropLibrary.getFactory().createDispatched(Constants.CacheSizes.BUILTIN_INTEROP_DISPATCH);
+  private @Child InteropLibrary library =
+      InteropLibrary.getFactory().createDispatched(Constants.CacheSizes.BUILTIN_INTEROP_DISPATCH);
 
-    private @Child
-    CoercePrimitiveNode coercion = CoercePrimitiveNode.build();
-    private final BranchProfile err = BranchProfile.create();
+  private @Child CoercePrimitiveNode coercion = CoercePrimitiveNode.build();
+  private final BranchProfile err = BranchProfile.create();
 
-    Object execute(Object array, long index) {
-        try {
-            Object elem = library.readArrayElement(array, index);
-            return coercion.execute(elem);
-        } catch (UnsupportedMessageException e) {
-            err.enter();
-            Builtins builtins = Context.get(this).getBuiltins();
-            throw new PanicException(
-                    builtins.error().makeTypeError(builtins.array(), array, "array"), this);
-        } catch (InvalidArrayIndexException e) {
-            err.enter();
-            Builtins builtins = Context.get(this).getBuiltins();
-            throw new PanicException(
-                    builtins.error().makeInvalidArrayIndexError(array, index), this);
-        }
+  Object execute(Object array, long index) {
+    try {
+      Object elem = coercion.execute(library.readArrayElement(array, index));
+      if (elem == null) {
```

Comment:
I see. `coercion` yields `null` and we replace it with `nothing().newInstance()`. Can't we do that at the moment of _returning null_? Not that it'd be performance problem, but we may forget to do the `null -> Nothing` conversion, right?

Reply:
So the main issue is that Epb knows nothing about Atoms and therefore I couldn't use `Nothing`

Reply:
See https://github.com/enso-org/enso/pull/3628#discussion_r940045334